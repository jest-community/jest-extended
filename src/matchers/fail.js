export function fail(_, message) {
  return {
    pass: false,
    message: () => (message ? message : 'fails by .fail() assertion'),
  };
}
