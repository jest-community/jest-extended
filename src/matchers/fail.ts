export function fail(_: unknown, message: string) {
  return {
    pass: false,
    message: () => (message ? message : 'fails by .fail() assertion'),
  };
}
