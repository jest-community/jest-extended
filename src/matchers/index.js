import toBeFalse from './toBeFalse';
import toBeTrue from './toBeTrue';

export default [toBeFalse, toBeTrue].reduce((acc, matcher) => ({ ...acc, ...matcher }), {});
