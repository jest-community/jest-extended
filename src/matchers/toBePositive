function toBePositive(val){
      if(Math.sign(val) <= -1){
        
       return 'please put a positive number';
      } else if(val == Infinity){
        return 'please put a positive number';
      }else if(isNaN(val)){
        return 'please put a positive number';
      }
      return val + ' is a positive Number';
}
  
  
  
toBePositive(1);
