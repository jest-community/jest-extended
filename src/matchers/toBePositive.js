export function toBePositive(actual) {
  const { printReceived, matcherHint } = this.utils;

  const passMessage =
    matcherHint('.not.toBePositive', 'received', '') +
    '\n\n' +
    'Expected value to not be positive received:\n' +
    `  ${printReceived(actual)}`;

  const failMessage =
    matcherHint('.toBePositive', 'received', '') +
    '\n\n' +
    'Expected value to be positive received:\n' +
    `  ${printReceived(actual)}`;

  const pass = actual !== true && !isNaN(actual) && actual !== Infinity && actual > 0;

  return { pass, message: () => (pass ? passMessage : failMessage) };
}
