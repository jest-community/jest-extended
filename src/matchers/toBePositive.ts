export function toBePositive(actual: unknown) {
  // @ts-expect-error OK to have implicit any for this
  const { printReceived, matcherHint } = this.utils;

  const pass = isNumber(actual) && isPositive(actual);

  return {
    pass,
    message: () =>
      pass
        ? matcherHint('.not.toBePositive', 'received', '') +
          '\n\n' +
          'Expected value to not be positive received:\n' +
          `  ${printReceived(actual)}`
        : matcherHint('.toBePositive', 'received', '') +
          '\n\n' +
          'Expected value to be positive received:\n' +
          `  ${printReceived(actual)}`,
  };
}

const isNumber = (value: any) => !isNaN(parseInt(value));
const isPositive = (value: any) => value > 0;
