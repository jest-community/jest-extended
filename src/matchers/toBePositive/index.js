import { matcherHint, printReceived } from 'jest-matcher-utils';

import predicate from './predicate';

const passMessage = received => () =>
  matcherHint('.not.toBePositive', 'received', '') +
  '\n\n' +
  'Expected value to not be positive received:\n' +
  `  ${printReceived(received)}`;

const failMessage = received => () =>
  matcherHint('.toBePositive', 'received', '') +
  '\n\n' +
  'Expected value to be positive received:\n' +
  `  ${printReceived(received)}`;

export default {
  toBePositive: expected => {
    const pass = predicate(expected);
    if (pass) {
      return { pass: true, message: passMessage(expected) };
    }

    return { pass: false, message: failMessage(expected) };
  }
};
