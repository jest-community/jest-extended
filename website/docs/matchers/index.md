---
sidebar_position: 1
---

# Matchers

- [.pass(message)](/docs/matchers/pass)
- [.fail(message)](/docs/matchers/fail)
- [.toBeEmpty()](/docs/matchers/tobeempty)
- [.toBeOneOf([members])](/docs/matchers/tobeoneof)
- [.toBeNil()](/docs/matchers/tobenil)
- [.toSatisfy(predicate)](/docs/matchers/tosatisfy)

## [Array](/docs/matchers/array)

- [.toBeArray()](/docs/matchers/array/#tobearray)
- [.toBeArrayOfSize()](/docs/matchers/array/#tobearrayofsize)
- [.toIncludeAllMembers([members])](/docs/matchers/array/#toincludeallmembersmembers)
- [.toIncludeAllPartialMembers([members])](/docs/matchers/array/#toincludeallpartialmembersmembers)
- [.toIncludeAnyMembers([members])](/docs/matchers/array/#toincludeanymembersmembers)
- [.toIncludeSameMembers([members])](/docs/matchers/array/#toincludesamemembersmembers)
- [.toPartiallyContain(member)](/docs/matchers/array/#topartiallycontainmember)
- [.toSatisfyAll(predicate)](/docs/matchers/array/#tosatisfyallpredicate)
- [.toSatisfyAny(predicate)](/docs/matchers/array/#tosatisfyanypredicate)
- [.toBeInRange(min, max)](/docs/matchers/array/#tobeinrangemin-max)

## [Boolean](/docs/matchers/boolean)

- [.toBeBoolean()](/docs/matchers/boolean/#tobeboolean)
- [.toBeTrue()](/docs/matchers/boolean/#tobetrue)
- [.toBeFalse()](/docs/matchers/boolean/#tobefalse)

## [Date](/docs/matchers/date)

- [.toBeDate()](/docs/matchers/date/#tobedate)
- [.toBeValidDate()](/docs/matchers/date/#tobevaliddate)
- [.toBeAfter(date)](/docs/matchers/date/#tobeafterdate)
- [.toBeBefore(date)](/docs/matchers/date/#tobebeforedate)
- [.toBeAfterOrEqualTo(date)](/docs/matchers/date/#tobeafterorequaltodate)
- [.toBeBeforeOrEqualTo(date)](/docs/matchers/date/#tobebeforeorequaltodate)
- [.toBeBetween(startDate, endDate)](/docs/matchers/date/#tobebetweenstartdate-enddate)

## [Function](/docs/matchers/function)

- [.toBeFunction()](/docs/matchers/function/#tobefunction)
- [.toThrowWithMessage()](/docs/matchers/function/#tothrowwithmessagetype-message)

## [Mock](/docs/matchers/mock)

- [.toHaveBeenCalledBefore()](/docs/matchers/mock/#tohavebeencalledbefore)
- [.toHaveBeenCalledAfter()](/docs/matchers/mock/#tohavebeencalledafter)
- [.toHaveBeenCalledOnce()](/docs/matchers/mock/#tohavebeencalledonce)
- [.toHaveBeenCalledOnceWith()](/docs/matchers/mock/#tohavebeencalledoncewith)

## [Number](/docs/matchers/number)

- [.toBeNumber()](/docs/matchers/number/#tobenumber)
- [.toBeNaN()](/docs/matchers/number/#tobenan)
- [.toBeFinite()](/docs/matchers/number/#tobefinite)
- [.toBePositive()](/docs/matchers/number/#tobepositive)
- [.toBeNegative()](/docs/matchers/number/#tobenegative)
- [.toBeEven()](/docs/matchers/number/#tobeeven)
- [.toBeOdd()](/docs/matchers/number/#tobeodd)
- [.toBeWithin(start, end)](/docs/matchers/number/#tobewithinstart-end)
- [.toBeInteger()](/docs/matchers/number/#tobeinteger)

## [Object](/docs/matchers/object)

- [.toBeObject()](/docs/matchers/object/#tobeobject)
- [.toBeEmptyObject()](/docs/matchers/object/#tobeemptyobject)
- [.toContainKey(key)](/docs/matchers/object/#tocontainkeykey)
- [.toContainKeys([keys])](/docs/matchers/object/#tocontainkeyskeys)
- [.toContainAllKeys([keys])](/docs/matchers/object/#tocontainallkeyskeys)
- [.toContainAnyKeys([keys])](/docs/matchers/object/#tocontainanykeyskeys)
- [.toContainValue(value)](/docs/matchers/object/#tocontainvaluevalue)
- [.toContainValues([values])](/docs/matchers/object/#tocontainvaluesvalues)
- [.toContainAllValues([values])](/docs/matchers/object/#tocontainallvaluesvalues)
- [.toContainAnyValues([values])](/docs/matchers/object/#tocontainanyvaluesvalues)
- [.toContainEntry([key, value])](/docs/matchers/object/#tocontainentrykey-value)
- [.toContainEntries([[key, value]])](/docs/matchers/object/#tocontainentrieskey-value)
- [.toContainAllEntries([[key, value]])](/docs/matchers/object/#tocontainallentrieskey-value)
- [.toContainAnyEntries([[key, value]])](/docs/matchers/object/#tocontainanyentrieskey-value)
- [.toBeExtensible()](/docs/matchers/object/#tobeextensible)
- [.toBeFrozen()](/docs/matchers/object/#tobefrozen)
- [.toBeSealed()](/docs/matchers/object/#tobesealed)

## [Promise](/docs/matchers/promise)

- [.toResolve()](/docs/matchers/promise/#toresolve)
- [.toReject()](/docs/matchers/promise/#toreject)

## [String](/docs/matchers/string)

- [.toBeString()](/docs/matchers/string/#tobestring)
- [.toBeHexadecimal(string)](/docs/matchers/string/#tobehexadecimal)
- [.toBeDateString(string)](/docs/matchers/string/#tobedatestringstring)
- [.toEqualCaseInsensitive(string)](/docs/matchers/string/#toequalcaseinsensitivestring)
- [.toStartWith(prefix)](/docs/matchers/string/#tostartwithprefix)
- [.toEndWith(suffix)](/docs/matchers/string/#toendwithsuffix)
- [.toInclude(substring)](/docs/matchers/string/#toincludesubstring)
- [.toIncludeRepeated(substring, times)](/docs/matchers/string/#toincluderepeatedsubstring-times)
- [.toIncludeMultiple([substring])](/docs/matchers/string/#toincludemultiplesubstring)
- [.toEqualIgnoringWhitespace(string)](/docs/matchers/string/#toequalignoringwhitespacestring)

## [Symbol](/docs/matchers/symbol)

- [.toBeSymbol()](/docs/matchers/symbol/#tobesymbol)
