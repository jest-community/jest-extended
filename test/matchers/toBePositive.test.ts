import * as matcher from 'src/matchers/toBePositive';

expect.extend(matcher);

describe('.toBePositive', () => {
  test('passes when given a positive number', () => {
    expect(1).toBePositive();
  });

  test('passes when given a positive BigInt', () => {
    expect(1n).toBePositive();
  });

  test('fails when not given a positive number', () => {
    expect(() => expect(-1).toBePositive()).toThrowErrorMatchingSnapshot();
  });

  test('fails when not given a positive BigInt', () => {
    expect(() => expect(-1n).toBePositive()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBePositive', () => {
  test.each([[false], [''], [-1], [0], [{}], [[]], [() => {}], [undefined], [null], [NaN], [Infinity], [-1n], [0n]])(
    'passes when not given a positive number: %s',
    given => {
      expect(given).not.toBePositive();
    },
  );

  test('fails when given a positive number', () => {
    expect(() => expect(5).not.toBePositive()).toThrowErrorMatchingSnapshot();
  });

  test('fails when given a positive BigInt', () => {
    expect(() => expect(5n).not.toBePositive()).toThrowErrorMatchingSnapshot();
  });
});
