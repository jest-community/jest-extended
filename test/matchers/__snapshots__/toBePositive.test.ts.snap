// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBePositive fails when given a positive number 1`] = `
"<dim>expect(</intensity><red>received</color><dim>).not.toBePositive()</intensity>

Expected value to not be positive received:
  <red>5</color>"
`;

exports[`.toBePositive fails when not given a positive number 1`] = `
"<dim>expect(</intensity><red>received</color><dim>).toBePositive()</intensity>

Expected value to be positive received:
  <red>-1</color>"
`;
